packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -345,6 +346,61 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             }
         }
 
+        private static IReadOnlyList<ParameterProvider> BuildOrderedSpreadParameters(InputModelType inputModel)
+        {
+            // create the model provider
+            if (ClientModelPlugin.Instance.TypeFactory.CreateModel(inputModel) is not ModelProvider modelProvider)
+                throw new InvalidOperationException($"Unable to create model for input model {inputModel.Name}.");
+
+            var fullConstructor = modelProvider.FullConstructor;
+            var primaryConstructor = modelProvider.Constructors.FirstOrDefault(c => c != fullConstructor);
+            var orderedParameters = new List<ParameterProvider>();
+            var primaryCtorParameterNames = new HashSet<string>();
+
+            // add the primary constructor parameters
+            if (primaryConstructor?.Signature.Parameters != null)
+            {
+                foreach (var param in primaryConstructor.Signature.Parameters)
+                {
+                    orderedParameters.Add(param);
+                    primaryCtorParameterNames.Add(param.Name);
+                }
+            }
+
+            // add the remaining constructor parameters from the full constructor, avoiding duplicates
+            foreach (var param in fullConstructor.Signature.Parameters)
+            {
+                var wireInfo = param.Property?.WireInfo;
+                if (wireInfo is null || primaryCtorParameterNames.Contains(param.Name))
+                    continue;
+
+                if (!wireInfo.IsRequired)
+                {
+                    orderedParameters.Add(new(
+                        param.Name,
+                        param.Description,
+                        param.Type.InputType,
+                        Default,
+                        param.IsRef,
+                        param.IsOut,
+                        [],
+                        param.Property,
+                        param.Field,
+                        param.InitializationValue)
+                    {
+                        Validation = param.Validation
+                    });
+
+                    continue;
+                }
+
+                orderedParameters.Add(param);
+            }
+
+            // sort the parameters by required
+            return [.. orderedParameters.OrderBy(p => p.DefaultValue == null ? 0 : 1)];
```

Comment:
I don't think this is the ordering we want.   We should follow these standards https://azure.github.io/azure-sdk/dotnet_implementation.html#parameter-presence-and-ordering.  This seems to arbitrarily mix headers, queries, etc.